### ISSUE TITLE
Magento 2 Grunt - Error arguments missing ')'

### ISSUE THREAD

**Issue by hiteshseepossible** [seanCodes/bootstrap-less-port#24]
I have tried to implement `bootstrap-less-port` into Magento 2. I have use **Grunt** for the compile less to CSS.

Installed Grunt Details.
> grunt-CLI v1.3.2
> grunt v1.0.4

### Error
> ParseError: Could not parse call arguments or missing ')' in pub/static/frontend/XXXXXX/XXXXX/en_US/css/source/less/_root.less on line 6, column 16:
> 5    //}
> 6    each(@colors, #(@value, @color) {
> 7            --@{color}: @value;

Image: https://prnt.sc/rfalir

#### Debugged
you have used `each(@list, #(@value) {})` means

```
each(@colors, #(@value, @color) {
    --@{color}: @value;
});
```
If we use without a defined variable.maybe it works. Ex. 
```
each(@colors,  {
    --@{key}: @value;
});
```
Can you please help for fix this issuse.

**Comment by seanCodes** [comment]
Hi @hiteshseepossible, can you tell me what version of Less you’re using?

**Comment by seanCodes** [comment]
Yeah it looks like the version of `grunt-contrib-less` that you’re using uses an older version of Less. Are you able to update the Magento dev dependencies? If so you could bump `grunt-contrib-less` to 2.0.0.

Alternatively, you could `npm install bootstrap-less-port@1.0.0`, since 1.0.0 is compatible with older versions of Less.

Let me know if either of those works for you or if you run into any more trouble.

**Comment by hiteshseepossible** [comment]
Thank you for the suggestion
But now facing another issue.
**Error**

>> pub/static/frontend/XXXX/en_US/css/source/less/_functions.less: [L10:C0] '_plugins/map-keys.js_' wasn't found. Tried - pub/static/frontend/XXXX/en_US/css/source/less/plugins/map-keys.js,pub/static/frontend/XXXX/en_US/css/plugins/map-keys.js,npm://plugins/map-keys.js,plugins/map-keys.js
Warning: Error compiling pub/static/frontend/XXXX/en_US/css/styles-m.less Use --force to continue.
>  **Aborted due to warnings.**

**Comment by seanCodes** [comment]
How are you importing bootstrap-less-port into your project? Also, could you do `npm ls --depth 0 | grep less` and tell me what  Less version is being used now?

**Comment by hiteshseepossible** [comment]
I have up your less folder into my directory. 
`npm ls --depth 0 | grep less`
got below output.
>├── grunt-contrib-less@1.4.1
npm ERR! peer dep missing: ajv@^5.0.0, required by ajv-keywords@2.1.1

**Comment by seanCodes** [comment]
Ok so you’re not importing from `node_modules` like this?
```less
@import “bootstrap-less-port”;
```

Also I messed up that command, can you tell me the output of `npm ls | grep less`

**Comment by hiteshseepossible** [comment]
NO, I have use direct your less folder. Can you please guide me on how to set up it, I am PHP dev :-)

`npm ls | grep less`

> ├--┬ grunt-contrib-less@2.0.0
    │ ├--┬ less@3.11.1
    │ │ ├---- caseless@0.12.0

>npm ERR! peer dep missing: ajv@^5.0.0, required by ajv-keywords@2.1.1

**Comment by hiteshseepossible** [comment]
I will explain, What Magento 2 did when I run Grunt.

We have to run below CLI for less.

`grunt exec`
It will copy all the less files into the `pub/static` folder. 
**NOTE:** but in this copied folder there is missing your plugin folder. 

`grunt less`
Conver those files into CSS.

**Comment by seanCodes** [comment]
Ah interesting, thank you for explaining.

I’m not sure why the plugin folder would not be copied, but maybe it’s because the plugin folder contains JS files? I’m not sure.

It sounds like this issue is out of my control, though. 🙁 Maybe you can look into the gruntfile and see what the `exec` task does and modify it to copy over all files (not just Less ones).

If you can link me to the gruntfile that contains the `exec` task I might be able to take a look at that code and help further. Otherwise you might need to create an issue in the Magento repo.

**Comment by hiteshseepossible** [comment]
You can check more in below link

GruntFile
https://github.com/magento/magento2/blob/2.4-develop/Gruntfile.js.sample

Setup Theme 
https://github.com/magento/magento2/blob/2.4-develop/dev/tools/grunt/configs/themes.js

we use below CLI for run

- Remove Cache
- Clean theme by the command: grunt clean
- Run CMD command prompt with administrator privilege.
- Run the command: grunt exec:yourthemename
- Run the command: grunt less:yourthemename
- Run the command: grunt watch

**Comment by mageho** [comment]
@hiteshseepossible as far as I know, Magento doesn't import (and support) less js plugin when you compile theme code, so it will always return errors.
You can do a workaround by using a recent less compiler by grunt command in developer mode
But when you will have to pass to production mode, it will not work.
A solution will be to download a downgraded version of this repository.

I am using a separate bootstrap.css file compiled from sass versions. But I lost all modularity.

I do not know if you know about "https://asset-packagist.org", you can using it in your magento project to download in a particular directory (I am using this one by example  "lib/web")
I was using it in developer mode to install bootstrap-less-port with Magento 2.
